plugins/services/src/js/reducers/serviceForm/Constraints.js

```diff
@@ -0,0 +1,125 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../src/js/constants/TransactionTypes';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+
+module.exports = {
+  JSONReducer(state, {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    if (this.constraints == null) {
+      this.constraints = [];
+    }
+
+    const joinedPath = path.join('.');
+
+    if (joinedPath.search('constraints') !== -1) {
+      if (joinedPath === 'constraints') {
+        switch (type) {
+          case ADD_ITEM:
+            this.constraints.push({field: null, operator: null, value: null});
+            break;
+          case REMOVE_ITEM:
+            this.constraints = this.constraints.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return this.constraints.filter((item) => {
+          return item.field != null && item.operator != null;
+        }).map(({field, operator, value}) => {
+          return value ? [field, operator, value] : [field, operator];
+        });
+      }
+
+      if (type !== SET) {
+        return state;
+      }
+
+      let index = joinedPath.match(/\d+/)[0];
+      let key = ['field', 'operator', 'value'].find(function (name) {
+        return `constraints.${index}.${name}` === joinedPath;
+      });
+      if (key !== -1) {
+        this.constraints[index][key] = value;
+      }
+    }
+
+    return this.constraints.filter((item) => {
+      return item.field != null && item.operator != null;
+    }).map(({field, operator, value}) => {
+      return value ? [field, operator, value] : [field, operator];
```

Comment:
We didn't decide to use the ternary operator yet. Let's hold off until the team has decided this.